runtime/assembly/__tests__/inherent.spec.ts

```diff
@@ -1,14 +1,13 @@
-import { Inherent } from '../models';
+import { InherentData } from '../models';
 import { MockBuilder } from './mock-builder';
 import { MockResult } from './mock-result';
 
 describe("Inherent", () => {
     it("Should instanciate new Inherent class from the empty SCALE Encoded Array", () => {
-        const mock: MockResult<Inherent> = MockBuilder.getInherentMock();
-        const decodedData = Inherent.fromU8Array(mock.bytes);
-
-        expect(decodedData == mock.instance).toBeTruthy();
+        const mock: MockResult<InherentData> = MockBuilder.getInherentMock();
+        const decodedData = InherentData.fromU8Array(mock.bytes);
+        assert(decodedData.result == mock.instance, "Expected object is not equal to the newly instanciated object");
         __retain(changetype<usize>(mock));
         __retain(changetype<usize>(decodedData));
     })
-})
\ No newline at end of file
+})
```

Comment:
It would be useful to add a test for the encoding, using the same `getInheretMock` function you will be able to do that very easily.

```JavaScript
it("should encode InherentData correctly", () => {
    const inherentData = MockBuilder.getInherentMock();
    assert(Utils.areArraysEqual(inherentData.instance.toU8a(), inherentData.bytes), "InherentData was not encoded successfully");

    __retain(changetype<usize>(inherentData));
  });
```